tiatoolbox/visualization/tileserver_api.yml

```diff
@@ -0,0 +1,399 @@
+        - in: path
+          name: res
+          description: Resolution of the tile (1 or 2x)
+          required: true
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: Image returned successfully
+          content:
+            image/webp:
+              schema:
+                type: image/webp
+
+  /:
+    get:
+      summary: Get index page
+      description: This endpoint serves a web page with a simple slide viewer
+      responses:
+        '200':
+          description: The index page.
+          content:
+            text/html:
+              schema:
+                type: string
+
+  /tileserver/session_id:
+    get:
+      summary: Get session_id information
+      description: Set up new session_id identifier to disambiguate different sessions
+      responses:
+        '200':
+          description: Successful response
+          headers:
+          Set-Cookie:
+            schema:
+              type: string
+            description: The 'session_id' cookie containing the session_id identifier.
+
+  /tileserver/color_prop:
+    put:
+      summary: Update the property used to determine the colors of rendered annotations
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                prop:
+                  type: string
+                  description: The property to color annotations by
+      responses:
+        '200':
+          description: Successful response
+    get:
+      summary: Get the property used to color annotations from renderer.
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: string
+
+  /tileserver/slide:
+    put:
+      summary: change the base slide. All overlays will be removed.
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                slide_path:
+                  type: string
+                  description: Path to the slide in url safe form
+      responses:
+        '200':
+          description: Successful response
+    get:
+      summary: Get the slide metadata.
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+
+  /tileserver/cmap:
+    put:
+      summary: Change colormap for annotation layer
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                cmap:
+                  type: string
+                  description: Name of the colormap, should be a matplotlib colormap or a dict string
```

Comment:
@measty should this be 'urlencoded' if it's not in the URL?

Reply:
I'm not sure familiar with the openapi yaml schema so I may be wrong but it just didn't look quite right.